server/jetstream_api.go

```diff
@@ -1614,7 +1631,9 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, subject, repl
 		} else if sa == nil {
 			if js.isLeaderless() {
 				resp.Error = jsClusterNotAvailErr
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+				// Delaying an error response gives the leader a chance to respond before us
+				strMsg := string(msg)
```

Comment:
You did not have to do this, could pass string(msg) directly to sendDelayedAPIErrResponse() since that function is not itself invoked in a different go routine, it only invoke something in a go routine.

Reply:
fixed